src/core/utils/misc.cpp

```diff
@@ -257,14 +257,14 @@ QString Utils::Misc::friendlyUnit(qreal val, bool is_speed)
 {
     if (val < 0)
         return QCoreApplication::translate("misc", "Unknown", "Unknown (size)");
+
     int i = 0;
-    while(val >= 1024. && i++<6)
+    int units_size = 5; // the size of units[]
+    for (; (val >= 1024.) && (i < units_size); ++i) {
         val /= 1024.;
-    QString ret;
-    if (i == 0)
-        ret = QString::number((long)val) + " " + QCoreApplication::translate("misc", units[0].source, units[0].comment);
-    else
-        ret = Utils::String::fromDouble(val, 1) + " " + QCoreApplication::translate("misc", units[i].source, units[i].comment);
+    }
+
+    QString ret = Utils::String::fromDouble(val, 1) + " " + QCoreApplication::translate("misc", units[i].source, units[i].comment);
```

Comment:
This is wrong because for bytes it will always print a decimal digit even if it is never possible to have fractinal bytes eg 15.5 Bytes.
So just change 6 to 5 in the loop and leave the rest as-is.